crates/templates/src/manager.rs

```diff
@@ -91,10 +124,23 @@ impl TemplateManager {
 
         let dest_dir = self.store.get_directory(&id);
         if dest_dir.exists() {
-            return Ok(InstallationResult::Skipped(
-                id.to_owned(),
-                SkippedReason::AlreadyExists,
-            ));
+            match options.exists_behaviour {
+                ExistsBehaviour::Skip => {
+                    return Ok(InstallationResult::Skipped(
+                        id.to_owned(),
+                        SkippedReason::AlreadyExists,
+                    ))
+                }
+                ExistsBehaviour::Update => tokio::fs::remove_dir_all(&dest_dir)
```

Comment:
This is somewhat unpleasant if the install ends up failing...

Reply:
Yeah, by this point everything is ready locally so it's just a local file copy which _should_ be reasonably safe... but you're right, we should try to make failure non-impactful.  I'll take a look.

Reply:
I've pushed something that reduces the window of error but I'm not sure if there's a better way to offer the guarantee.  One possibility is to indirect through an index file: then we can simply install the new version to a completely fresh directory and update the index (and then garbage collect any unreferenced directories).  There may be value in an index file anyway (e.g. to simplify searching) - let me know if you think this is the way to go, or if you have other suggestions.  Thanks!

Reply:
I was going to suggest printing a description of what went wrong to allow manual cleanup, but this works too!